text/0003-measure-metric-type.md

```diff
@@ -2,50 +2,169 @@
+
+To guide the user in selecting the right kind of metric for an application, we'll consider the following questions about the primary intent of reporting given data.  We use "of primary interest" here to mean information that is almost certainly useful in understanding system behavior.  Consider these questions:
+
+- Does the measurement represent a quantity of something?  Is it also non-negative?
+- Is the sum a matter of primary interest?
+- Is the event count a matter of primary interest?
+- Is the distribution (p50, p99, etc.) a matter of primary interest?
+
+The specification will be updated with the following guidance.
+
+### Cumulative metric
+
+Likely to be the most common kind of metric, cumulative metric events express the computation of a sum.  Choose this kind of metric when the value is a quantity, the sum is of primary interest, and the event count and distribution are not of primary interest.  To raise (or lower) a cumulative metric, call the `Add()` method.
+
+If the quantity in question is always non-negative, it implies that the sum is strictly ascending.  When this is the case, the cumulative metric also serves to define a rate.  For this reason, cumulative metrics have a `NonNegative` option to be declared as non-negative.  The API will reject negative updates to non-negative cumulative metrics, instead submitting an SDK error event, which helps ensure meaningful rate calculations.
+
+For cumulative metrics, the default OpenTelemetry implementation exports the sum of event values taken over an interval of time.
+
+### Gauge metric
+
+Gauge metrics express a pre-calculated value that is either `Set()` by explicit instrumentation or observed through a callback.  Generally, this kind of metric should be used when the metric cannot be expressed as a sum or a rate because the measurement interval is arbitrary.  Use this kind of metric when the measurement is not a quantity, and the sum and event count are not of interest.
+
+Only the gauge kind of metric supports observing the metric via a gauge `Observer` callback (as an option).  Semantically, there is an important difference between explicitly setting a gauge and observing it through a callback.  In case of setting the gauge explicitly, the call happens inside of an implicit or explicit context.  The implementation is free to associate the explicit `Set()` event with a context, for example.  When observing gauge metrics via a callback, there is no context associated with the event.
+
+As a special case, to support existing metrics infrastructure, a gauge metric may be declared as a precomputed cumulative sum using the `NonDescending` option, in which case it is defined as a strictly ascending.  The API will reject descending updates to non-descending gauges, instead submitting an SDK error event.
```

Comment:
Same here about the strictness of the ascending.

Also, should it be clarified whether the `Observer` callback is called if gauge is set to the same value as before?

Reply:
There's now a separate RFC about the `Observer`, see https://github.com/open-telemetry/oteps/pull/39.

But, the pattern is reversed -- the Observer will be called to get the current value when the SDK wants it. It will be up to the metric exporters to decide how to handle zero-updates in the output. My preference is to report 0-change events on gauges, it makes a difference in some products.